CONTRIBUTING.md

```diff
@@ -158,3 +223,319 @@ including how to setup a local Kubernetes cluster, can be found in
+
+Before you begin work on your feature it is import to ensure that you have
+proper review and approval from the project maintainers.
+
+Examples of a new feature include:
+* Adding a new set of metrics for enabling additional telemetry.
+* Adding additional supported transport layer protocol options for network policy.
+* Adding support for IPSec.
+
+#### Deprecation
+
+A `kind/deprecation` label categorizes an issue or PR as related to feature
+marked for deprecation.
+
+To create a deprecation issue:
+
+* label your issue with `kind/deprecation`
+* title the issue with the feature you are deprecating
+* describe the deprecation in the issue body making sure to:
+  * explain why the feature is being deprecated
+  * discuss time-to-live for the feat
+  * discuss any impacts to existing APIs
+
+#### Design
+
+A `kind/design` label categorizes issue or PR as related to design.
+
+A design issue is for discussing larger architectural and design proposals.
+Approval of a design proposal may result in multiple additional feature,
+api-change, or cleanup issues being created to implement the design.
+
+To create a design issue:
+
+* label your issue with `kind/design`
+* describe the design in the issue body
+
+Before creating additional issues that implement the proposed design it is
+important to get feedback and approval from the maintainers. Design feedback
+could include some of the following:
+
+* needs additional detail
+* no, this problem should be solved in another way
+* this is desireable but we need help completing other issues first; then we will
+  consider this design
+
+_Note this was previously called `proposal`. We will add the `kind/design` label
+to all issues currently labeled with `proposal`._
+
+#### Documentation
+
+A `kind/documentation` label categorizes issue or PR as related to a
+documentation.
+
+To create a documentation issue:
+
+* label your issue with `kind/documentation`
+* title the issue with a short description of what you are documenting
+* provide a brief summary in the issue body of what you are documenting. In some
+  cases, it might be useful to include a checklist of changed documentation
+  files to indicate your progress.
+
+#### Failing Test
+
+A `kind/failing-test` label categorizes issue or PR as related to a consistently
+or frequently failing test.
+
+To create a failing test issue:
+
+* label your issue with `kind/failing-test`
+
+TODO: As more automation is used in the continuous integration pipeline, we will
+be able to automatically generate an issue for failing tests.
+
+#### Support
+
+A `kind/support` label categorizes issue as related to a support request.
+
+To create a support issue:
+
+* label your issue with `kind/support`
+* title the issue with a short description of your support request
+* answer all of the questions in the support issue template
+
+_Note this was previously called `support`. We will add the `kind/support` label
```

Comment:
same comment as above

also doesn't render as intended because of missing trailing underscore